File: 5-Problem-A-Day/34. Find First and Last Position of Element in Sorted Array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] ans = {-1, -1};
        int left = 0, right = nums.length;
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] < target)
                left = mid + 1;
            else
                right = mid;
        }
        if (left == nums.length || nums[left] != target)
            return ans;
        ans[0] = left;
        left = 0;
        right = nums.length;
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] > target)
                right = mid;
            else
                left = mid + 1;
        }
        ans[1] = left - 1;
        return ans;
    }
}

File: 2023/34. Find First and Last Position of Element in Sorted Array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int left = 0, right = (int) nums.size();
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] < target){
                left = mid + 1;
} else {
    right = mid;
}
        }
        if (left >= nums.size() || nums[left] > target) {
            return {-1, -1};
}
        int first = left;
        left = 0, right = (int)nums.size();
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] <= target) {
               left = mid + 1;
} else {
      right = mid;
            }
}
   return {first, left - 1};
    }
};